freeipa/src/main/java/com/sequenceiq/freeipa/sync/ProviderChecker.java

```diff
@@ -77,6 +79,28 @@
         }, LOGGER, ":::Auto sync::: provider is checked in {}ms");
     }
 
+    public Set<InstanceMetaData> updateStatuses(Stack stack, InstanceStatus instanceStatus, boolean updateStatusFromFlow) {
```

Comment:
- `updateStatusFromFlow` doesn't make sense here as it is called only from one place
- I don't think this is the right class for this method as this method doesn't check anything on the provider side
- please cover it with unit tests

Reply:
Thx! I moved the method to a separate InstanceUpdater.java class which is similar to the StackUpdater.java class. Now it is more clear.

Also, I created unit tests to cover it.